nilearn/image/image.py

```diff
@@ -311,6 +328,190 @@ def smooth_img(imgs, fwhm):
+
+    Examples
+    --------
+    >>> from nilearn import datasets, surface
+    >>> curv = datasets.load_fsaverage_data(data_type="curvature")
+    >>> curv_smooth = surface.smooth_img(curv, iterations=50)
+
+    """
+    # First, calculate the center and surround weights for the
+    # center-surround knob.
+    center_weight = 1 / (1 + np.exp2(-center_surround_knob))
+    surround_weight = 1 - center_weight
+    if surround_weight == 0:
+        # There's nothing to do in this case.
+        return imgs
+
+    # Calculate the adjacency matrix either weighting
+    # by inverse distance or not weighting (ones)
+    values = "invlen" if distance_weights else "ones"
+
+    _ = _sanitize_surface_weights(imgs, vertex_weights=vertex_weights)
+
+    new_data = {}
+    for hemi, n_iter in zip(imgs.mesh.parts, iterations):
+        mesh = imgs.mesh.parts[hemi]
+        data = imgs.data.parts[hemi]
+
+        if n_iter == 0:
+            new_data[hemi] = data
+            continue
+
+        matrix = compute_adjacency_matrix(mesh, values=values)
+
+        # We need to normalize the matrix columns, and we can do this now by
+        # normalizing everything but the diagonal to the surround weight, then
+        # adding the center weight along the diagonal.
+        colsums = matrix.sum(axis=1)
+        colsums = np.asarray(colsums).flatten()
+        matrix = matrix.multiply(surround_weight / colsums[:, None])
+        # Add in the diagonal.
+        matrix.setdiag(center_weight)
+
+        # Run the iterations of smoothing.
+        tmp = data
+        for _ in range(n_iter):
+            tmp = matrix.dot(tmp)
+
+        # Convert back into numpy array.
+        new_data[hemi] = np.reshape(np.asarray(tmp), np.shape(data))
+
+    smoothed_imgs = new_img_like(imgs, new_data)
+
+    return smoothed_imgs
+
+
+def _mris_fwhm_to_niters(fwhm, img) -> list[int]:
+    """Convert a desired FWHM to number of smoothing iterations for surface.
+
+    Adapted from freesurfer
+    https://github.com/freesurfer/freesurfer/tree/main/utils/mrisutils.cpp#L1100
+
+    Parameters
+    ----------
+    fwhm : :obj:`float`
+        Full width at half maximum (in mm)
+
+    img : surface image
+
+    Returns
+    -------
+    niters: list of number of smoothing iterations (one per mesh in the image)
+    """
+    if fwhm is None:
+        fwhm = 0
+    # Convert FWHM to standard deviation of Gaussian kernel
+    G_STD = fwhm / math.sqrt(math.log(256.0))
+
+    niters = []
+    for mesh in img.mesh.parts.values():
+        if fwhm == 0:
+            niters.append(0)
+            continue
+
+        if isinstance(mesh, FileMesh):
+            mesh = mesh.loaded()
+
+        # Compute average vertex area
+        avg_vertex_area = mesh._area / mesh.n_vertices
+
+        # Compute number of iterations using empirical formula
+        niters.append(
+            math.floor(
+                1.14 * (4 * math.pi * G_STD**2) / (7 * avg_vertex_area) + 0.5
+            )
+        )
```

Comment:
Is there any way to check this formula ? E.g. smoothing a dirac on a sphere, for which we can probably predict the expected outcome?